src/test/srcShared/fs.test.ts

```diff
@@ -0,0 +1,230 @@
+/*!
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as assert from 'assert'
+import * as vscode from 'vscode'
+import * as path from 'path'
+import { existsSync, mkdirSync, readFileSync, rmSync, writeFileSync } from 'fs'
+import { FakeExtensionContext } from '../fakeExtensionContext'
+import fsCommon from '../../srcShared/fs'
+import * as os from 'os'
+import { isMinimumVersion } from '../../shared/vscode/env'
+
+function isWin() {
+    return os.platform() === 'win32'
+}
+
+describe('FileSystem', function () {
+    let fakeContext: vscode.ExtensionContext
+
+    before(async function () {
+        fakeContext = await FakeExtensionContext.create()
+    })
+
+    beforeEach(async function () {
+        await makeTestRoot()
+    })
+
+    afterEach(async function () {
+        await deleteTestRoot()
+    })
+
+    describe('readFileAsString()', function () {
+        it('reads a file', async function () {
+            const path = await makeFile('test.txt', 'hello world')
+            const content = await fsCommon.readFileAsString(path)
+            assert.strictEqual(content, 'hello world')
+        })
+
+        it('reads a file from a URI', async function () {
+            const path = await makeFile('test.txt', 'hello world')
+            const uri = vscode.Uri.file(path)
+            const content = await fsCommon.readFileAsString(uri)
+            assert.strictEqual(content, 'hello world')
```

Comment:
Could cover the string/URI variants in one test. That saves a lot of code (given all the string/URI variants in fs.ts), which allows the reader to focus on the more interesting and relevant things to be tested.

The variants are probably not even needed, since passing a string implicitly tests both (and the typescript type system tells us the possible inputs).
```suggestion
            const content = await fsCommon.readFileAsString(uri)
            assert.strictEqual(content, 'hello world')
            assert.strictEqual((await fsCommon.readFileAsString(path)), 'hello world')
```

Reply:
Makes sense. I'll just keep this single test that takes the URI input as a sanity check, but will merge the existing test in to one.